returnn/torch/data/pipeline.py

```diff
@@ -463,6 +466,66 @@ def __getitem__(self, index):
         raise Exception(f"{self.__class__.__name__}.__getitem__ not supported")
 
 
+class ShufflingDataPipe(torch.utils.data.IterDataPipe):
```

Comment:
This does not really belong into this PR?

Reply:
Ah now I see, you rewrote that. That's also a solution. But it's different to what I suggested. Why do you do this now and not what I suggested?

Reply:
I know how I can make the values monotonic w/ another adapter, but I think that w/o having access to all batches in the shuffle buffer ensuring monotonicity will result in jumps and "skips" in the values (since you effectively do a `batch["complete_frac"] = max(last_complete_frac, batch["complete_frac"]`). By re-writing the adapter we can keep the same, smooth, monotonic values that we have had before. I don't know if there is a better way other than re-writing the shuffler. Do you know a better way?

Reply:
Yes sure. I just thought that the monotonicity-ensuring-adapter is more generic and might be useful in other cases as well. But yes, what you did is better for this specific case.

I just wanted to make sure there was no misunderstanding. And I was wondering initially why you did it.

I think it's fine and it can stay this way.